### PR TITLE
Support specific the klog device

### PR DESCRIPTION
**Pull request by tiann**
if the `/dev/kmsg` is not exists, users can create it manually by `mknod <device_path> c 1 11`

This PR make users can pass the actual kmsg device

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -59,29 +59,47 @@ pub struct KernelLog {
...
+        Self::with_device_and_level(Self::DEFAULT_DEVICE, filter)
+    }
+
+    /// Create new kernel logger from specific device
+    pub fn with_device(device: &str) -> io::Result<KernelLog> {
+        Self::with_device_and_level(device, LevelFilter::Trace)
+    }
+
+    /// Create new kernel logger from specific device with error level filter
+    pub fn with_device_and_level(device: &str, filter: LevelFilter) -> io::Result<KernelLog> {
```
Comment (kstep) [inline src/lib.rs]: Shouldn't the device type be `&Path` or `impl AsRef<Path>` (which is even better)?

**Comment by kstep** [comment]
@tiann LGTM, except for one minor neat-pick. Please consider the comment.

File: src/lib.rs (hunk 1 above)
Comment (tiann) [reply src/lib.rs]: maybe `impl AsRef<Path>` is better, as the std lib always use this.

**Comment by tiann** [comment]
> @tiann LGTM, except for one minor neat-pick. Please consider the comment.

thank you, i've fixed it :)

**Pull request merged**